apps/src/lib/util/AnalyticsReporter.js

```diff
@@ -0,0 +1,134 @@
+import {
+  init,
+  track,
+  Identify,
+  identify
+  // setUserId
+} from '@amplitude/analytics-browser';
+import {currentLocation} from '@cdo/apps/utils';
+
+// A flag that can be toggled to send events regardless of environment
+const ALWAYS_PUT = false;
+
+const Environments = {
+  production: 'production',
+  levelbuilder: 'levelbuilder',
+  test: 'test',
+  staging: 'staging',
+  adhoc: 'adhoc',
+  development: 'development',
+  unknown: 'unknown'
+};
+
+class AnalyticsReporter {
+  constructor() {
+    // Get the API key from the DOM. See application.html.haml layout.
+    const element = document.querySelector('script[data-amplitude-api-key]');
+    const api_key = element ? element.dataset.amplitudeApiKey : '';
+    // Let Amplitude handle missing/invalid keys.
+    if (this.shouldPutRecord(ALWAYS_PUT)) {
+      init(api_key);
+    }
+  }
+
+  setUserProperties(userId, userType, signInState) {
+    const identifyObj = new Identify();
+    const formattedUserId = this.formatUserId(userId);
+    // Dont set user id pending privacy conversations
+    // setUserId(formattedUserId);
```

Comment:
Is this the part you mentioned about privacy concerns around userid? Just wondering why it's commented out.

Reply:
Yay!!